### ISSUE TITLE
List of authors in commentary function is not limited to authors with granted access

### ISSUE THREAD

**Issue by nmueller18** [fiduswriter/fiduswriter#1037]
On FidusWriter.com, the commentary function (see screenshot, not showing the actual authors for privacy reasons) shows all authors the user has worked with in the past, not only those which have access granted to this particular document. Can this be correct?
As a sidenote, there is a glitch in the German localization, as "Änderungsverfolgung" (see again screenshot) is longer than the button.

![FW-screenshot](https://user-images.githubusercontent.com/20249289/87833758-eddccf00-c888-11ea-90d3-703d57404d8a.png)

**Comment by johanneswilm** [comment]
Maybe we should turn the German translation back to what it was? Or would you have a suggeston for a shorter version of it that actually fits the size of the button? "Än.ver."?

**Comment by johanneswilm** [comment]
and yes, this was done on purpose. The reason was that some users want to tag their collaborators in a bunch of places BEFORE they actually give them access to the document.

**Comment by johanneswilm** [comment]
I've tried to improve the German translation in 3.8.19.

**Comment by nmueller18** [comment]
> I've tried to improve the German translation in 3.8.19.

"Änderungen" was exactly what I wanted to suggest ...

**Comment by nmueller18** [comment]
> and yes, this was done on purpose. The reason was that some users want to tag their collaborators in a bunch of places BEFORE they actually give them access to the document.

I did not realize that the author-list is user-specific. I thought that it is document-specific and therefore suspected a security/privacy-issue.
I can see why someone wants it that way but it still does feel odd. At least, I would expect that the list of authors differs from that of collaborators. In the end, one can only add comments with already granted access to the document. That way a user would also know who is already author and who "only" former collaborator from an older project.
At some point (when closing the document?), the user should perhaps receive a warning that certain collaborators have been assigned tasks but not granted access to deal with them.

**Comment by johanneswilm** [comment]
I agree with several of the points you made above. For the next version, it would be good to think a bit about how one could improve the situation. There will however likely always be some of this oddity: if a comment is assigned to a user who subsequently deletes his/her account, we likely don't want the comment to just disappear. So some comments will always be assigned to users who do not currently have access to the document. But maybe a warning like the one you mention would make sense.

**Comment by nmueller18** [comment]
> if a comment is assigned to a user who subsequently deletes his/her account, we likely don't want the comment to just disappear.

I tend to disagree. If a user deliberately chooses to delete his/her account, I think s/he also has a right to disappear from the documents altogether (this even could be an obligation on the side of FidusWriter but I am not a lawyer). Perhaps his/her comments could then be reassigned to a generic user "Unknown"?
What about the situation if the owner of a document revokes grant access of another user? Perhaps this should not be possible as long as this user has active comments?

**Comment by johanneswilm** [comment]
>  I think s/he also has a right to disappear from the documents altogether (this even could be an obligation on the side of FidusWriter but I am not a lawyer).

What is stored is the username which the collaborator typed or clicked on - so there should be no personal information beyond that. And that username not something the user with that username entered, but something another user entered. So if the comment was previously displayed using first and last name of a given user, that first and last name is replaced with the username when the user has been deleted.

If a user collaborates on a document and then subsequently deletes their account, would you expect us to delete all their contributions in that document? That would potentially leave the remaining part of the document as gibberish. 

What do you do if the access to a given document is temporarily revoked and then restored again? if we remove all comments when one user's access is revoked or the user is pausing their account and when the user then regains access all their comments are gone? I would think that is probably not a good idea.

I am not a lawyer either, but as far as I can tell, GDPR does not require us to do this.

**Comment by nmueller18** [comment]
> If a user collaborates on a document and then subsequently deletes their account, would you expect us to delete all their contributions in that document? That would potentially leave the remaining part of the document as gibberish.
> 
> What do you do if the access to a given document is temporarily revoked and then restored again? if we remove all comments when one user's access is revoked or the user is pausing their account and when the user then regains access all their comments are gone? I would think that is probably not a good idea.

Convinced. I now see that this can become quite messy.

However, I still think that it would be helpful to differentiate in the comments between "Authors" – individuals who already have contributed to the document in question in the past (regardless if as still active users or with/without access to the document) – and "Collaborators" – Authors + potential contributors in the future. So, the first list would be generated on a document-specific, the second on a user-specific basis.

Related to this, are there plans to extend a similar functionality to track changes? In some/most word processors with track changes it is possible to filter changes by author. I think this is a useful feature.
